src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -273,6 +273,35 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
+        internal IEnumerable<TopicPartitionOffset> OffsetsForTimes(IEnumerable<TopicPartitionTimestamp> timestampsToSearch, int millisecondsTimeout)
+        {
+            var offsets = timestampsToSearch.Select(t => new TopicPartitionOffset(t.TopicPartition, t.Timestamp.UnixTimestampMs)).ToList();
+            IntPtr cOffsets = GetCTopicPartitionList(offsets);
+            try
+            {
+                // The timestamps to query are represented as Offset property in offsets param on input, 
+                // and Offset property will contain the offset on output
+                var errorCode = LibRdKafka.offsets_for_times(handle, cOffsets, (IntPtr) millisecondsTimeout);
+                if (errorCode != ErrorCode.NoError)
+                {
+                    throw new KafkaException(errorCode);
+                }
+
+                var list = GetTopicPartitionOffsetErrorList(cOffsets);
+                var error = list.Where(x => x.Error.Code != ErrorCode.NoError).Select(x => x.Error).FirstOrDefault();
```

Comment:
When I suggested making a `TopicPartitionTimestamp` type, I wasn't realizing that `offsets_for_times` could return per partition errors. Sorry! We should not have the `TopicPartitionTimestamp` type, replace it with a `TopicPartitionTimestampError` type and not throw an exception when a partition specific error occurs.